### PR TITLE
feat(rff-19): creating a stepper

### PR DESCRIPTION
**Pull request by jisung-lee7**
## Issue Link

[rff-19](https://linear.app/coding-for-fun/issue/RFF-19/create-a-stepper)

## Notes

-

### CODE REVIEW

**Comment by jisung-lee7** [comment]
I'm not sure yet what else needs to be added to the Stepper. 
Therefore, I'll implement React Hook Form, additional props, and logic as I move forward with the next steps(If it need).

File: src/elements/stepper/stepper.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            {currentStep !== cureentStepLastIndex ? (
+              <Button onClick={handleToNext}>Next</Button>
+            ) : (
+              <Button onClick={handleToSubmit}>Submit</Button>
+            )}
+          </div>
+        </>
+      ) : (
+        <p>Invalid step</p>
+      )}
```
Comment (jiyeol-lee) [inline src/elements/stepper/stepper.tsx:54]: You don't have to think about the invalid step. This is the biggest mistake that front-end engineers can make. This is just a component and if the developer decide to pass invalid step, that's developer's fault. And think what could happen to the user? Nothing. The most important thing is data (database) and passing the right data to the backend, not micro-managing of the front-end side state.

File: src/elements/stepper/stepper.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          </div>
+          <div>
+            <Button onClick={handleToPrevious} disabled={currentStep === 0}>
+              Previous
+            </Button>
+            {currentStep !== cureentStepLastIndex ? (
+              <Button onClick={handleToNext}>Next</Button>
+            ) : (
+              <Button onClick={handleToSubmit}>Submit</Button>
+            )}
```
[the remark this answers is not in the document]
Comment (jisung-lee7) [reply src/elements/stepper/stepper.tsx:49]: Could you provide additional explanation? I don’t understand what I need to do and why it’s necessary.

[the remark this answers is not in the document]
Comment (jiyeol-lee) [reply src/elements/stepper/stepper.tsx]: Umm, never mind what I said. Now I'm thinking we shouldn't even have these buttons in this component. We are creating a stepper component not something else. We probably not need this buttons when we are using stepper. Or we could need something else over the button. Basically it's completely depending on the business requirement. So I think it's redundant to add these buttons inside this stepper component.

[the remark this answers is not in the document]
Comment (jisung-lee7) [reply src/elements/stepper/stepper.tsx]: I don't know exactly but yes sir. I removed it.

[the remark this answers is not in the document]
Comment (jiyeol-lee) [reply src/elements/stepper/stepper.tsx]: Nope, tell me what you don't know.

[the remark this answers is not in the document]
Comment (jisung-lee7) [reply src/elements/stepper/stepper.tsx]: Here is my thought process. 
let’s assume we first display the repository tree. 
Then, with the selected pulls, I think a button is necessary to proceed to step 2. 
If there’s anything I’m missing or not considering further, please let me know.

[the remark this answers is not in the document]
Comment (jiyeol-lee) [reply src/elements/stepper/stepper.tsx]: Yes, you are right. I didn't mean that we don't need button. We need button but that button should not stay inside this component. It should be passed from the parent component.

There could be two ways.

1. Just pass action prop and render it inside the Stepper component.

```html
<Stepper
  currentStep={step}
  steps={steps}
  handleStepClick={handleStepClick}
  actions={
    <Button>Prev</Button>
    <Button>Next</Button>
  }
/>
```

2. Render separate from the parent component
```html
<div class="flex flex-col gap-3>
  <Stepper
    currentStep={step}
    steps={steps}
    handleStepClick={handleStepClick}
  />
  <div>
    body...
  </div>
  <div class="flex justify-end items-center gap-3">
    <Button>Prev</Button>
    <Button>Next</Button>
  </div>
</div>
```

I'm kind of leaning to option 2.

[the remark this answers is not in the document]
Comment (jisung-lee7) [reply src/elements/stepper/stepper.tsx]: I got how to handle the buttons.
Then can you explain the reason for separating the button like this? 
I’m wondering why I feel that the button should be part of the stepper component.

[the remark this answers is not in the document]
Comment (jiyeol-lee) [reply src/elements/stepper/stepper.tsx]: It's simple. What we want to do make is `stepper` component, not `stepper` + `button` + `body`. If you are dumping more and more in one place, that core component will be getting bigger and bigger and you won't be able to understand your own code.

[the remark this answers is not in the document]
Comment (jisung-lee7) [reply src/elements/stepper/stepper.tsx]: I got it. I feel like I lack experience(짬빠) in this area. Thank you.

File: src/elements/stepper/stepper.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  currentStep: number
+  steps: Stepper[]
+}
+
+export const Stepper = ({ currentStep, steps }: IStepperProps) => {
+  return (
+    <div>
+      <>
+        <div>
+          <h3>Step {steps[currentStep]!.value}</h3>
```
Comment (jiyeol-lee) [inline src/elements/stepper/stepper.tsx:16]: Let's pass this one as a prop. It could be `title` prop

Comment (jisung-lee7) [reply src/elements/stepper/stepper.tsx]: @jiyeol-lee uhm,, bro like it?

Comment (jiyeol-lee) [reply src/elements/stepper/stepper.tsx]: Nope, it's not `children`. Let's name it `title`

Comment (jisung-lee7) [reply src/elements/stepper/stepper.tsx]: uhm,, is it right?

Comment (jiyeol-lee) [reply src/elements/stepper/stepper.tsx]: Yes

**Pull request merged**